internal/handlers/commonparams/params.go

```diff
@@ -93,36 +95,49 @@ func getWholeParamStrict(command string, param string, value any) (int64, error)
 			)
 		case errors.Is(err, ErrNotWholeNumber):
 			if math.Signbit(value.(float64)) {
-				return 0, commonerrors.NewCommandError(
+				return 0, commonerrors.NewCommandErrorMsgWithArgument(
 					commonerrors.ErrValueNegative,
-					fmt.Errorf("BSON field '%s' value must be >= 0, actual value '%d'", param, int(math.Ceil(value.(float64)))),
+					fmt.Sprintf(
+						"BSON field '%s' value must be >= %d, actual value '%d'",
+						param, boundary, int(math.Ceil(value.(float64))),
+					),
+					param,
```

Comment:
I'm not sure what we should set for currently `scale` is param or `key` from `setStructField(...)`. In other places, we use `$collStats (stage)`, `$gt` and `projection`. I think command name is preferred over parameter name. With saying that, parameter name is technically argument so not sure myself.

> NewCommandErrorMsgWithArgument creates a new wire protocol error with an argument that caused the error.

Reply:
Yeah, we can do that. Changed to `command`.